RUFAS/input_manager.py

```diff
@@ -53,144 +77,368 @@ def _load_metadata(self, metadata_path: str = "input/example_metadata.json") ->
-        om.add_log("Total Invalid Critical Elements", f"{invalid_critical_elements_counter=}", info_map)
-
-        if invalid_critical_elements_counter > 0:
-            return False
+                        return False
 
-        return True
+        om.add_log("Total Valid Items", f"{valid_elements_counter=}", info_map)
+        om.add_log("Total Checked Items", f"{total_elements_counter=}", info_map)
+        om.add_log("Total Fixed Items", f"{fixed_elements_counter=}", info_map)
+        om.add_log("Total Invalid Items", f"{invalid_elements_counter=}", info_map)
+        return invalid_elements_counter == 0
 
-    def _validate_element(self, key: str, value: Any) -> bool:
+    def _validate_element(self, element_hierarchy: List[str], properties_blob_key: str,
+                          input_data: Dict[str, Any], eager_termination: bool,
+                          counter_dict: Dict[str, int], ) -> bool:
         """
         Perform data validation checks.
 
         Parameters
         ----------
-        key : str
-            The key of the data to validate.
+        element_hierarchy : List[str]
+            A list of strings representing the path to the data being validated.
 
-        value : Any
-            The value of the data to validate.
+        properties_blob_key : str
+            The metadata properties section keyword for the data input file being checked.
+
+        input_data : Dict[str, Any]
+            A buffer dictionary that holds the input data for validation and fixing.
+
+        eager_termination : bool
+            If true, the process will be terminated upon finding invalid data.
 
+        counter_dict : Dict[str, int]
+            A dictionary that collects the counts of total elements checked,
+            invalid elements, valid elements, and fixed elements.
 
         Returns
         -------
         bool
             True if the data is valid, False otherwise.
         """
-        # Perform data validation checks
-        # Return True if the data is valid, False otherwise
+        info_map = {"class": self.__class__.__name__,
+                    "function": self._validate_element.__name__,
+                    }
+        variable_properties = reduce(lambda d, key: d[key], element_hierarchy,
+                                     self.__metadata["properties"][properties_blob_key])
+        var_type = variable_properties["type"]
+        is_nested = var_type == "object"
+        if is_nested:
+            children_status: Dict[str, bool] = {}
+            false_counter = 0
+            for nested_key in variable_properties.keys():
+                element_hierarchy.append(nested_key)
+                child_status = self._validate_element(element_hierarchy, properties_blob_key,
+                                                      input_data, eager_termination)
+                if eager_termination and not child_status:
```

Comment:
rename it in line 234 as well
```suggestion
                if eager_termination and not is_child_valid:
```

Reply:
updated